docs/source/explanations/dataclass.md

```diff
@@ -0,0 +1,455 @@
+So you *must* include a type annotation for your field to be included in the
+`__init__` signature.  Variables without a default value are considered
+mandatory
+
+```python
+In [1]: from dataclasses import dataclass
+
+In [2]: @dataclass
+    ...: class Test:
+    ...:     x: int  # mandatory parameter
+    ...:     y: str = 'hi'  # optional parameter
+    ...:     z = 2  # no type annotation, NOT in __init__
+    ...:
+
+In [3]: Test?
+Init signature: Test(x: int, y: str = 'hi') -> None
+Docstring:      Test(x: int, y: str = 'hi')
+```
+
+### Handy `dataclass` features
+
+The python documentation for `dataclass` is good, and relatively short, so we
+won't duplicated it here, but here are some handy features that may be
+particularly useful when defining napari models:
+
+1. Add custom [**post-init
+   processing**](https://docs.python.org/3/library/dataclasses.html#post-init-processing)
+   by declaring a `__post_init__` method on your class.
+2. [**Exclude specific
+   variables**](https://docs.python.org/3/library/dataclasses.html#class-variables)
+   from dataclass fields by annotating as `typing.ClassVar`.
+3. Declare [**init-only
+   variables**](https://docs.python.org/3/library/dataclasses.html#init-only-variables)
+   (variables that are passed to `__post_init__` but are otherwise *not* counted
+   as fields in the dataclass) by annotating as type `dataclasses.InitVar`.
+4. Provide a [**`default_factory`
+   function**](https://docs.python.org/3/library/dataclasses.html#default-factory-functions)
+   to generate the default value for a field using a function (this is important
+   for [mutable default
+   values](https://docs.python.org/3/library/dataclasses.html#mutable-default-values)
+   such as an empty list.)
+    ```python
+    from dataclasses import dataclass, field
+
+    @dataclass
+    class D:
+        x: list = field(default_factory=list)
+    ```
+5. Dataclasses *can* be [used as a base class for
+   **inheritance**](https://docs.python.org/3/library/dataclasses.html#inheritance),
+   but know that there are some tricky scenarios; specifically: the subclass
+   cannot have any non-defaut arguments if the parent class defines any fields
+   with defaults. (see [this stack-overflow
+   post](https://stackoverflow.com/questions/51575931/class-inheritance-in-python-3-7-dataclasses)
+   for details)
+
+In some cases, you may also need to modify the behavior of the full dataclass, or of individual fields in the dataclass using the `init`, `repr`, `order`, `unsafe_hash`, or `frozen` parameters as well, so it's worth [reviewing their function](https://docs.python.org/3/library/dataclasses.html#dataclasses.dataclass).
+
+## napari's `evented_dataclass`
+
+The `@evented_dataclass` decorator adds two more parameters to the standard library `dataclass`: `events` and `properties`.  **By default, they are both `True`.**
+
+### `@evented_dataclass(events=True)`
```

Comment:
`events=True` is default right? it might more confusing to see that called out if not necessary, but i don't mind it either